### PR TITLE
isl: add isl/0.23 + bump deps + set pkg_config name

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **isl/0.23**

- [ ] Depends on #3836 

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by mathbunnyru** [comment]
@madebr could you please trigger this recipe once again?

https://github.com/conan-io/conan-center-index/pull/4137/files
I think I have issues here, which were already fixed by this PR.

**Review by mathbunnyru — APPROVED** [review]

File: recipes/isl/all/conanfile.py [hunk 1]
```diff
@@ -39,54 +39,61 @@ def configure(self):
         del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
 
+        if self.settings.os == "Windows" and self.options.shared:
```
Comment (uilianries) [inline recipes/isl/all/conanfile.py:42]: it can be promoted to validate() method:

https://blog.conan.io/2020/12/14/New-conan-release-1-32.html

**Review by uilianries — APPROVED** [review]
LGTM

**Review by SpaceIm — APPROVED** [review]

File: recipes/isl/all/conanfile.py (hunk 1 above)
Comment (prince-chrismc) [reply recipes/isl/all/conanfile.py:42]: This is rarely being done and I needs a hook... is there a related issue? I don't see it https://github.com/conan-io/hooks/issues

**Review by prince-chrismc — APPROVED** [review]

File: recipes/isl/all/conanfile.py (hunk 1 above)
Comment (Croydon) [reply recipes/isl/all/conanfile.py:42]: It is too new and there might be valid cases where you want to throw earlier, not sure if a hook is the answer

Comment (SpaceIm) [reply recipes/isl/all/conanfile.py:42]: Unless a check requires to access the final version/options values of upstreams, I don't see the advantage to check in `validate()`

Comment (prince-chrismc) [reply recipes/isl/all/conanfile.py:42]: It's starting to get messy when/why to use this new feature... we need to be more vigilant when reviewing

**Pull request merged**